app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt

```diff
@@ -920,23 +1000,78 @@ private fun assertStartBrowsingButton() {
 }
 
 // Pick your toolbar placement
-private fun assertTakePlacementHeader() {
+private fun assertToolbarPlacementHeader() {
     scrollToElementByText(STRING_ONBOARDING_TOOLBAR_PLACEMENT_HEADER)
     onView(allOf(withText(STRING_ONBOARDING_TOOLBAR_PLACEMENT_HEADER)))
         .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
 }
 
-private fun assertTakePacementTopRadioButton() {
-    scrollToElementByText(STRING_ONBOARDING_TOOLBAR_PLACEMENT_HEADER)
-    onView(ViewMatchers.withResourceName("toolbar_top_radio_button"))
-        .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
-}
+private fun assertToolbarPlacementDescription() =
+    assertTrue(
+        mDevice
+            .findObject(
+                UiSelector()
+                    .textContains(
+                        getStringResource(R.string.onboarding_toolbar_placement_description),
+                    ),
+            ).waitForExists(waitingTime),
+    )
 
-private fun assertTakePlacementBottomRadioButton() {
-    scrollToElementByText(STRING_ONBOARDING_TOOLBAR_PLACEMENT_HEADER)
-    onView(ViewMatchers.withResourceName("toolbar_bottom_radio_button"))
-        .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
-}
+private fun assertToolbarPlacementTopRadioButton(isChecked: Boolean) =
+    assertTrue(
+        mDevice
+            .findObject(
+                UiSelector()
+                    .resourceId("$packageName:id/toolbar_top_radio_button")
+                    .checked(isChecked),
+            ).waitForExists(waitingTime),
+    )
+
+private fun assertToolbarPlacementBottomRadioButton(isChecked: Boolean) =
+    assertTrue(
+        mDevice
+            .findObject(
+                UiSelector()
+                    .resourceId("$packageName:id/toolbar_bottom_radio_button")
+                    .checked(isChecked),
+            ).waitForExists(waitingTime),
+    )
+
+private fun assertToolbarPlacementBottomImage() =
+    assertTrue(
+        mDevice
+            .findObject(
+                UiSelector()
+                    .resourceId("$packageName:id/toolbar_bottom_image"),
+            ).waitForExists(waitingTime),
+    )
+
+private fun assertToolbarPlacementTopImage() =
+    assertTrue(
+        mDevice
+            .findObject(
+                UiSelector()
+                    .resourceId("$packageName:id/toolbar_top_image"),
+            ).waitForExists(waitingTime),
+    )
+
+private fun assertToolbarPlacementBottomTitle() =
+    assertTrue(
+        mDevice
+            .findObject(
+                UiSelector()
+                    .resourceId("$packageName:id/toolbar_bottom_title"),
+            ).waitForExists(waitingTime),
+    )
+
+private fun assertToolbarPlacementTopTitle() =
+    assertTrue(
+        mDevice
+            .findObject(
+                UiSelector()
+                    .resourceId("$packageName:id/toolbar_top_title"),
+            ).waitForExists(waitingTime),
+    )
```

Comment:
I'm starting to worry of the size of these robots and all these single use functions. Can we loop through a single function (e.g, verifyToolbarUI) that passes in the resource? (A pattern to start using in other places too)

Reply:
Sounds good @AaronMT , did the same for clicking and long clicking different web page items.
I'll go ahead and refactor where applicable.

Reply:
Made the refactoring, please share your thoughts @AaronMT and @sv-ohorvath.